cheatsheets_draft/GraphQL_Cheat_Sheet.md

```diff
@@ -0,0 +1,302 @@
+
+It's possible for a GraphQL API to support access to objects using their ID even if that is not intended. Sometimes there are `node` or `nodes` or both fields in a query object, and these can be used to access objects directly by `ID`. You can check whether your schema has these fields by running this on the command line (assuming that `schema.json` contains your GraphQL schema): `cat schema.json | jq ".data.__schema.types[] | select(.name==\"Query\") | .fields[] | .name" | grep node`. Removing these fields from the schema should disable the functionality, but you should always apply proper authorization checks to verify the caller has access to the object they are requesting.
+
+#### Query Access (Data Fetching)
+
+As part of a GraphQL API there will be various data fields that can be returned. One thing to consider is if you want different levels of access around these fields. For example, you may only want certain consumers to be able to fetch certain data fields rather than allowing all consumers to be able to retrieve all available fields. This can be done by adding a check in the code to ensure that the requester should be able to read a field they are trying to fetch.
+
+#### Mutation Access (Data Manipulation)
+
+GraphQL supports mutation, or manipulation of data, in addition to its most common use case of data fetching. If an API implements/allows mutation then there may need to be access controls put in place to restrict which consumers, if any, can modify data through the API. Setups that require mutation access control would include APIs where only read access is intended for requesters or where only certain parties should be able to modify certain fields.
+
+***Disable Introspection - Java***
+
+```Java
+GraphQLSchema schema = GraphQLSchema.newSchema()
+    .query(StarWarsSchema.queryType)
+    .fieldVisibility( NoIntrospectionGraphqlFieldVisibility.NO_INTROSPECTION_FIELD_VISIBILITY )
+    .build();
+```
+
+***Disable Introspection & GraphiQL - JavaScript***
+
+```javascript
+app.use('/graphql', graphqlHTTP({
+  schema: MySessionAwareGraphQLSchema,
++ validationRules: [NoIntrospection]
+  graphiql: process.env.NODE_ENV === 'development',
+}));
+```
+
+### Batching Attacks
```

Comment:
This is the only section that is brand new. Would love a look from you @PauloASilva